src/types/UiaTextRangeBase.cpp

```diff
@@ -990,1534 +904,364 @@ const COORD UiaTextRangeBase::_getScreenFontSize() const
-                                                  const std::wstring_view wordDelimiters,
-                                                  _Out_ gsl::not_null<int*> const pAmountMoved)
-{
-    *pAmountMoved = 0;
-    ScreenInfoRow currentScreenInfoRow = 0;
-    Column currentColumn = 0;
-
-    // set current location vars
-    if (endpoint == TextPatternRangeEndpoint::TextPatternRangeEndpoint_Start)
-    {
-        currentScreenInfoRow = moveState.StartScreenInfoRow;
-        currentColumn = moveState.StartColumn;
-    }
-    else
-    {
-        currentScreenInfoRow = moveState.EndScreenInfoRow;
-        currentColumn = moveState.EndColumn;
-    }
-
-    auto& buffer = pData->GetTextBuffer();
-    for (int i = 0; i < abs(moveCount); ++i)
-    {
-        // check if we're at the edge of the screen info buffer
-        if (currentScreenInfoRow == moveState.LimitingRow &&
-            currentColumn == moveState.LastColumnInRow)
-        {
+            nextPos.X = bufferSize.RightInclusive();
+            fSuccess = bufferSize.IncrementInBounds(nextPos, allowBottomExclusive);
+            if (fSuccess)
+            {
+                resultPos = nextPos;
+                (*pAmountMoved)++;
+            }
             break;
         }
-        else if (currentColumn == moveState.LastColumnInRow)
+        case MovementDirection::Backward:
         {
-            // we're at the edge of a row and need to go to the
-            // next one. move to the cell with the last non-whitespace charactor
-
-            currentScreenInfoRow += static_cast<int>(moveState.Increment);
-            // get the right cell for the next row
-            const ROW& row = buffer.GetRowByOffset(currentScreenInfoRow);
-            const size_t right = row.GetCharRow().MeasureRight();
-            currentColumn = gsl::narrow<Column>((right == 0) ? 0 : right - 1);
-
-            // get the right-most word for the previous row
-            const auto point = _screenInfoRowToEndpoint(pData, currentScreenInfoRow) + currentColumn;
-            auto target = _endpointToCoord(pData, point);
-
-            target = buffer.GetWordStart(target, wordDelimiters, true);
-
-            if (endpoint == TextPatternRangeEndpoint::TextPatternRangeEndpoint_Start)
+            // can't move past top
+            if (!allowBottomExclusive && nextPos.Y == bufferSize.Top())
             {
-                buffer.GetSize().IncrementInBounds(target);
+                fSuccess = false;
+                break;
             }
 
-            currentColumn = target.X;
-        }
-        else
-        {
-            // moving somewhere away from the edges of a row
-            const Endpoint point = _screenInfoRowToEndpoint(pData, currentScreenInfoRow) + currentColumn;
-            auto target = _endpointToCoord(pData, point);
-            buffer.GetSize().DecrementInBounds(target);
+            // NOTE: Automatically detects if we are trying to move past origin

```

Comment:
The previous branch (move forward) sets X to the right side so that IncrementInBounds wraps to the next line.

This branch does _not_ set X to 0 so that DecrementInBounds wraps to the previous line.
Is that intended?

It seems like moving a char backward by unit line would not work because of this.

Reply:
oh, is moving backwards by a line supposed to move _to_ the beginning of the line? wow, non-obvious

Reply:
Yes. A successful move by line is always supposed to result in `X = Left()`.

For start, makes sense. We want it to be 0 so that we get the full line.

For end, it's weird. We need to be on `X=0` of the _next_ line. Having `X=Right*()` would not be right since end is exclusive. Adding a comment in the function